src/solver/impls/snes/snes.cxx

```diff
@@ -56,25 +92,40 @@ int SNESSolver::init(int nout, BoutReal tstep) {
   timestep =
       (*options)["timestep"].doc("Initial backward Euler timestep").withDefault(1.0);
 
+  dt_min_reset = (*options)["dt_min_reset"]
+                   .doc("If dt falls below this, reset to starting dt")
+                   .withDefault(1e-6);
+
   diagnose =
-      (*options)["diagnose"].doc("Print additional diagnostics").withDefault(false);
+      (*options)["diagnose"].doc("Print additional diagnostics").withDefault<bool>(false);
 
-  predictor = (*options)["predictor"].doc("Use linear predictor?").withDefault(true);
+  predictor =
+      (*options)["predictor"].doc("Use linear predictor?").withDefault<bool>(true);
 
+  equation_form = (*options)["equation_form"]
```

Comment:
Could `equation_form` be a `BOUT_ENUM_CLASS`? Then users could use the names directly, rather than integers